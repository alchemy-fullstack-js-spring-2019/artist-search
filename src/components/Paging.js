import React from 'react';
import PropTypes from 'prop-types';

function Paging({ currentPage, totalPages, changePage }) {
  return (
    <div style={{ display: 'flex', justifyContent: 'center', alignItems: 'center' }}>
      <button style={{ height: '20px' }} onClick={() => changePage(currentPage - 1)}> Prev </button>
      <p>{currentPage + 1} of {totalPages}</p>
      <button style={{ height: '20px' }} onClick={() => changePage(currentPage + 1)}> Next </button>
    </div>
  );
}

Paging.propTypes = {
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  changePage: PropTypes.func.isRequired
};

export default Paging;
