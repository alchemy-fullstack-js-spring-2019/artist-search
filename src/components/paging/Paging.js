import React from 'react';
import PropTypes from 'prop-types';

export default function Paging({ currentPage, totalPages, updatePage }) {
  return (
    <>
      <span>
        {currentPage} of {totalPages}
      </span>
      { currentPage > 1 && <button onClick={ () => updatePage(currentPage - 1) }>Previous Page</button> }
      { currentPage < totalPages && <button onClick={ () => updatePage(currentPage + 1) }>Next Page</button> }
    </>
  );
}

Paging.propTypes = {
  currentPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  updatePage: PropTypes.func.isRequired
};
