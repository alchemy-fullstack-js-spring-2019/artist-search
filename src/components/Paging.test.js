import React from 'react';
import { shallow } from 'enzyme';
import Paging from './Paging';

describe('Paging component', () => {
  it('matches snapshot', () => {
    const wrapper = shallow(<Paging currentPage={1} totalPages={2} changePage={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
