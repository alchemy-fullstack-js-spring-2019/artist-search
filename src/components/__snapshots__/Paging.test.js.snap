// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paging component matches snapshot 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <button
    onClick={[Function]}
    style={
      Object {
        "height": "20px",
      }
    }
  >
     Prev 
  </button>
  <p>
    2
     of 
    2
  </p>
  <button
    onClick={[Function]}
    style={
      Object {
        "height": "20px",
      }
    }
  >
     Next 
  </button>
</div>
`;
